sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -185,7 +196,13 @@ def encrypt(self, algorithm, plaintext, **kwargs):
             try:
                 return self._local_provider.encrypt(algorithm, plaintext)
             except Exception as ex:  # pylint:disable=broad-except
+                if self._local_only:
+                    raise
```

Comment:
Should we log the warning first?

Reply:
I wasn't sure if logging before throwing would be redundant, but it doesn't hurt to do both